### ISSUE TITLE
Remote IP Issue

### ISSUE THREAD

**Issue by fz9** [engintron/engintron#553]
CloudLinux 7.3 (cageFS, lsphp, lsapi)
cPanel 64.0 (Build 24)
Engintron v1.8.3

Apache is not getting the real visitor IP, instead it is showing server IP. Verdict is NO as per your ip.php script. Below are the things I have tried.

-- Used the update/reinstall engintron option from within the WHM engintron window.
-- Tried several times "Restore Nginx IP forwarding in Apache"
-- Performed the following commands:
> cp -ax /var/cpanel/templates/apache2_4/ea4_main.default /var/cpanel/templates/apache2_4/ea4_main.local
>
> sed -i 's/LogFormat "%h/LogFormat "%a/'  /var/cpanel/templates/apache2_4/ea4_main.local
>
> /scripts/rebuildhttpdconf 
>
> service httpd restart

-- New httpd config is generating with a %a
-- Tried yum remove "ea-apache24-mod_remoteip-2.4.25-9.el7.cloudlinux.3.x86_64" and then clicked "Restore Nginx IP forwarding in Apache"

Even after all the steps, its still a no go! Please help!

**Comment by bober2000** [comment]
The same problem in #545

**Comment by fz9** [comment]
@bober2000 the difference here from #545 is that I am getting a NO instead of YES. So, its not even passing the IP to apache.

**Comment by fz9** [comment]
@bober2000 the difference here from #545 is that I am getting a NO instead of YES. So, its not even passing the IP to apache.

**Comment by rapidcloubr** [comment]
I'm having the same problem, my clients need real IP.

**Comment by fz9** [comment]
I understand this is not a commercial application and that support shouldn't be taken for granted, but it has been 2 weeks since this was posted and I am yet to get a response from the developer. Not a good sign for us who are facing this issue daily on production servers. At the least, please acknowledge the issue. Thanks!

**Comment by rapidcloubr** [comment]
Hello,

Please, we need an update regarding the problem.

You could license ENGINTRON and provide better support. The Engintron is great, but it looks like the project stopped.

**Comment by fevangelou** [comment]
v1.8.4 resolved this. If you had modified your Apache configuration, then that could be problem.

Otherwise, you can repair logging from the WHM app (look at the Apache section).

**Comment by fz9** [comment]
@fevangelou Thank you for your reply, but I just updated to v1.8.4, and the issue still remains. I'm not sure what you mean my modified Apache config? Which part of the config are you referring to?

You do realize that the "Restore Nginx IP forwarding in Apache" just tries to install the Package "ea-apache24-mod_remoteip-2.4.25-9.el7.cloudlinux.3.x86_64" right? and that it does nothing if it says that the module is already installed? As stated above, this has already been done several times and it just states its already installed nothing to do.

Request you to not be vague on this and please state what config Apache is supposed to have for Enginton to make this to work properly? Hoping for a quick reply.

**Comment by fevangelou** [comment]
@fz9 That command will also patch your Apache config so log analyzers like AWStats can parse Apache's log files properly.

The best way to resolve this is to:
a) first check Nginx's access log. Does it show the correct user IP? If it does, then switch to checking Apache's domlogs.
b) make sure there is no other proxy in front of Nginx that could cause issues
c) make sure no other Nginx for cPanel plugin was previously installed
d) make sure that mod_remoteip does get installed. Simply examine the output of the "Restore Nginx IP forwarding in Apache" command.

As a sidenote, the sed command you run is in the wrong file. You should look at the actual httpd.conf file used by Apache.

**Comment by fz9** [comment]
@fevangelou 

a) Nginx access logs does show all the different user access IPs. However, Apache domlogs all show same IP i.e server IP
b) I only have Engintron installed and nothing else for proxy/caching
c) Engintron is the only Nginx plugin/proxy we have ever installed.
d) As indicated in the first post, when click on "Restore Nginx IP forwarding in Apache", it just says already installed, nothing to do. And as stated, it was manually removed and installed it again through the Restore IP forwarding option. Still didn't give IP to apache.

e) No, that is the correct file because cPanel overwrites the httpd.conf file with whatever is in main.default when apache gets rebuilt. And as I mentioned in the frist post, after running the sed command, the actual httpd.conf file did reflect the correct log format anyways. Even manually editing the httpd.conf file with that format still does nothing!

**Comment by fevangelou** [comment]
This file /var/cpanel/templates/apache2_4/ea4_main.local may be the template file used to build the Apache config but it's not the file Engintron modifies. The changes done to httpd.conf are also not just related to LogFormat. It also add an include statement to the related config for mod_remoteip:

`Include "/usr/local/apache/conf/includes/remoteip.conf"`

So if you modify the template file, you need to add this as well. In any case, examine the installer here: https://github.com/engintron/engintron/blob/master/engintron.sh#L39

Now, what does this output?
`cat /usr/local/apache/conf/includes/remoteip.conf`

If no such file can be found, try looking in /etc/apache2/conf.modules.d/ for a *_mod_remoteip.conf (where * is a number). If you find such a file, what's its contents?

**Comment by fz9** [comment]
@fevangelou 

`[root@xxx conf.modules.d]# cat 360_mod_remoteip.conf
# mod_remoteip (https://httpd.apache.org/docs/current/mod/mod_remoteip.html)
LoadModule remoteip_module modules/mod_remoteip.so
RemoteIPHeader        X-Forwarded-For
RemoteIPInternalProxy 127.0.0.1 10.0.0.95 10.0.0.111 10.0.0.112
[root@xxx conf.modules.d]#
`
the server is an EC2 instance on AWS, so it uses Private NAT IPs which are mapped to the public IP in cPanel.

**Comment by fevangelou** [comment]
These IPs however is what cPanel has stored. And what Engintron reads to setup the proxy configuration.

Modify this file and replace the 10.x IPs with the actual public IP.

So as you can see, you were the one being a bit vague as to your networking setup.

**Comment by fz9** [comment]
@fevangelou Thank you for helping to get to the bottom of it. I did not mean that as an accusation, I came here hoping to get help, and if I knew beforehand that this is where the config is stored, I would have changed it. I just wish we had this conversation a month ago when I opened the ticket because I was looking for specific instructions as not all of us are great Linux admins. However, I'm thankful for your help and having this resolved.
